src/components/select/Search.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from "react";
+import "./search.css";
+
+export const SearchableSelect = ({ options, onSelectFromList }) => {
+  const [filteredOptions, setFilteredOptions] = useState([options]);
+  const [selectedOption, setSelectedOption] = useState(null);
+  const [isListVisiable, setIsListVisiable] = useState(false);
+
+  const handleInputChange = (e) => {
+    const inputValue = e.target.value;
+    filterOptions(inputValue);
+  };
+
+  const filterOptions = (inputvalue) => {
+    const filteredOptions = options.filter(
+      (option) =>
+        option.label.toLowerCase().indexOf(inputvalue.toLowerCase()) > -1
+    );
+    setFilteredOptions(filteredOptions);
+  };
+
+  const handleOptionClick = (option) => {
+    setSelectedOption(option);
+    onSelectFromList(option.value);
+    setFilteredOptions([]);
+    handleShowList();
+  };
+
+  const handleShowList = () => {
+    setIsListVisiable((prevState) => (prevState === true ? false : true));
```

Comment:
This can be simplified like this

```suggestion
    setIsListVisiable((prevState) => !prevState);
```